### ISSUE TITLE
Add tools.vcvars_command support for amd64_x86 cross-toolchain in VS2013+

### ISSUE THREAD

**Issue by puetzk** [conan-io/conan#3157]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Conan 1.5.1, Windows 7, Visual Studio 2017

When building a package with
[settings]
arch=x86
arch_build=x86_64
compiler=Visual Studio
compiler.version=15

I expected tools.vcvars_command to select the amd64_x86 cross-toolchain. However, it selects the x86 native toolchain instead, ignoring arch_build. This means the compiler (and espescially linker) are more limited on available RAM, which can inhibit LTCG or other whole-program optimizations, or could even make the build fail for code with sufficiently horrifying template monstrosities from the depths of Boost (out of memory)

An amd64_x86 cross-toolchain is available starting with VS2013, so the first part of the fix would be to change

https://github.com/conan-io/conan/blob/a8649076fa6bfeb6c0b7438832fd996e56d72ecf/conans/client/tools/win.py#L244

to

```python
'x86': 'amd64_x86' if int(compiler_version) >= 12 else 'x86',
```

Which makes use of the amd64_x86 cross-toolchain on visual studio versions that support it, falling back to the current behavior for those that did not.

Secondly (but somewhat separately), I think it would make sense for this to follow settings.arch_build, rather than detected_architecture() (i.e. platform.machine()). This would give explicit control of the toolchain arch, and make it separate from the flavor of python being used to run conan. It would also be more consistent with tools.oss.get_cross_building_settings.

**Comment by claasd** [comment]
Unfortunately, implementing this ticket broke our silverlight builds.
Now, when a recipe does not support cross compiling and does not have arch_build in it's settings, since 1.7 amd64_x86 is chosen instead of x86.

**Comment by memsharded** [comment]
@claasd working on a fix for it for 1.9.1. Would that be good, or would you like a backport of the fix for 1.7.X too?

**Comment by claasd** [comment]
Hi, we have a workaround (String replace on the vcvars command), therefore a backport is not needed.
